internal/versions/versions.go

```diff
@@ -36,29 +36,102 @@ func SupportedK8sVersions() []string {
 	return validVersionsSorted
 }
 
+// SupportedValidK8sVersions returns a typed list of supported Kubernetes versions.
+func SupportedValidK8sVersions() (res []ValidK8sVersion) {
+	for _, v := range SupportedK8sVersions() {
+		res = append(res, ValidK8sVersion(v))
+	}
+	return
+}
+
 // ValidK8sVersion represents any of the three currently supported k8s versions.
 type ValidK8sVersion string
 
 // NewValidK8sVersion validates the given string and produces a new ValidK8sVersion object.
+// It accepts a major minor version (e.g. 1.26) and transforms it into a supported patch version (e.g. 1.26.7).
+// It also accepts a full version (e.g. 1.26.7) and validates it.
 // Returns an empty string if the given version is invalid.
 // strict controls whether the patch version is checked or not.
-// If strict is false, the patch version is ignored and the returned
-// ValidK8sVersion is a supported patch version for the given major.minor version.
+// If strict is false, the patch version is ignored and the returned.
 func NewValidK8sVersion(k8sVersion string, strict bool) (ValidK8sVersion, error) {
+	prefixedVersion := compatibility.EnsurePrefixV(k8sVersion)
+	parsedVersion, err := resolveK8sPatchVersion(prefixedVersion)
+	if err != nil {
+		return "", fmt.Errorf("resolving kubernetes patch version from flag: %w", err)
+	}
 	var supported bool
 	if strict {
-		supported = isSupportedK8sVersionStrict(k8sVersion)
+		supported = isSupportedK8sVersionStrict(parsedVersion)
 	} else {
-		supported = isSupportedK8sVersion(k8sVersion)
+		supported = isSupportedK8sVersion(parsedVersion)
 	}
 	if !supported {
-		return "", fmt.Errorf("invalid Kubernetes version: %s; supported versions are %v", k8sVersion, SupportedK8sVersions())
+		return "", fmt.Errorf("invalid Kubernetes version: %s; supported versions are %v", parsedVersion, SupportedK8sVersions())
+	}
+	return ValidK8sVersion(parsedVersion), nil
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (v *ValidK8sVersion) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var version string
+	if err := unmarshal(&version); err != nil {
+		return err
+	}
+	valid, err := NewValidK8sVersion(version, false) // allow any patch version to not force K8s patch upgrades
+	if err != nil {
+		return fmt.Errorf("unsupported Kubernetes version %s, supported versions are %s", version, strings.Join(SupportedK8sVersions(), ", "))
+	}
+	*v = valid
+	return nil
+}
+
+// resolveK8sPatchVersion takes the user input from --kubernetes and transforms a MAJOR.MINOR definition into a supported
+// MAJOR.MINOR.PATCH release.
+func resolveK8sPatchVersion(k8sVersion string) (string, error) {
```

Comment:
A function in `internal/` should not know about flags from `cli/`.
Either adjust this function to remove references to said flags, or move this function back into `cli/`